torch/csrc/cuda/Event.cpp

```diff
@@ -0,0 +1,179 @@
+}
+
+static struct PyGetSetDef THPVariable_properties[] = {
+  {"cuda_event", (getter)THPVariable_get_cuda_event, nullptr, nullptr, nullptr},
+  {nullptr}
+};
+
+static PyMethodDef THCPEvent_methods[] = {
+  {(char*)"record", (PyCFunction)THCPEvent_record, METH_O, nullptr},
+  {(char*)"wait", (PyCFunction)THCPEvent_wait, METH_O, nullptr},
+  {(char*)"query", (PyCFunction)THCPEvent_query, METH_NOARGS, nullptr},
+  {(char*)"elapsed_time", (PyCFunction)THCPEvent_elapsed_time, METH_O, nullptr},
+  {(char*)"synchronize",
+    (PyCFunction)THCPEvent_synchronize, METH_NOARGS, nullptr},
+  {(char*)"ipc_handle", (PyCFunction)THCPEvent_ipc_handle, METH_NOARGS, nullptr},
+  {nullptr}
+};
+
+PyTypeObject THCPEventType = {
+  PyVarObject_HEAD_INIT(nullptr, 0)
+  "torch._C._CudaEventBase",             /* tp_name */
+  sizeof(THCPEvent),                     /* tp_basicsize */
+  0,                                     /* tp_itemsize */
+  (destructor)THCPEvent_dealloc,         /* tp_dealloc */
+  0,                                     /* tp_print */
+  0,                                     /* tp_getattr */
+  0,                                     /* tp_setattr */
+  0,                                     /* tp_reserved */
+  0,                                     /* tp_repr */
+  0,                                     /* tp_as_number */
+  0,                                     /* tp_as_sequence */
+  0,                                     /* tp_as_mapping */
+  0,                                     /* tp_hash  */
+  0,                                     /* tp_call */
+  0,                                     /* tp_str */
+  0,                                     /* tp_getattro */
+  0,                                     /* tp_setattro */
+  0,                                     /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
+  nullptr,                                  /* tp_doc */
+  0,                                     /* tp_traverse */
+  0,                                     /* tp_clear */
+  0,                                     /* tp_richcompare */
+  0,                                     /* tp_weaklistoffset */
+  0,                                     /* tp_iter */
+  0,                                     /* tp_iternext */
+  THCPEvent_methods,                     /* tp_methods */
+  0,                                     /* tp_members */
+  THPVariable_properties,                /* tp_getset */
+  0,                                     /* tp_base */
+  0,                                     /* tp_dict */
+  0,                                     /* tp_descr_get */
+  0,                                     /* tp_descr_set */
+  0,                                     /* tp_dictoffset */
+  0,                                     /* tp_init */
+  0,                                     /* tp_alloc */
+  THCPEvent_pynew,                       /* tp_new */
+};
+
+bool THCPEvent_init(PyObject *module) {
```

Comment:
It's a bit better to make this `void THCPEvent_init (...)` and throw a `python_error` on failure because it will avoid hiding the specific underlying Python error messages with general failure messages.

Not all of the XXX_init functions follow this pattern, but I think it's good practice for new modules. Some follow this pattern, like THPSize_init.